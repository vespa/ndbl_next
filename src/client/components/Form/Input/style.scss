.input {

  width: 100%;
  line-height: 3rem;
  height: 3rem;
  border: 0;
  padding: 0 1rem;
}