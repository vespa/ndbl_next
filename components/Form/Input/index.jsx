import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './style.scss';
/* eslint-env browser */


const Input = ({
  title,
  name,
  type,
  required,
  validation,
  value,
  onChange,
}) => {
  return (
    <Fragment>
      <label htmlFor={name}>
        <span style={{ display: 'none' }}>
          {title}
        </span>
        <input
          type={type}
          name={name}
          id={name}
          required={required}
          title={title}
          placeholder={title}
          validation={validation}
          className={` ${style.input} inp inp-${name}`}
          value={value}
          onChange={onChange}
        />
      </label>
    </Fragment>
  );
};

Input.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  required: PropTypes.bool,
  validation: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

Input.defaultProps = {
  validation: '',
  required: false,
  value: '',
  onChange: () => {},
};
export default Input;
