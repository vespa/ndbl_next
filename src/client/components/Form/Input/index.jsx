import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './style.scss';
/* eslint-env browser */


const Input = ({
  title,
  name,
  type,
  required,
  validation,
  value,
  onChange,
  children,
}) => {
  return (
    <Fragment>
      <label htmlFor={name}>
        <span style={{ display: 'none' }}>
          {title}
        </span>
        <input
          type={type}
          name={name}
          id={name}
          required={required}
          title={title}
          placeholder={title}
          validation={validation}
          className={` ${style.input} inp inp-${name}`}
          value={value}
          onChange={onChange}
        />
        {children}
      </label>
    </Fragment>
  );
};

Input.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  required: PropTypes.bool,
  validation: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  children: PropTypes.node,
};

Input.defaultProps = {
  validation: '',
  required: false,
  value: '',
  children: <span />,
  onChange: () => {},
};
export default Input;
