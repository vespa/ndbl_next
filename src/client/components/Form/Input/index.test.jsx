import React from 'react';
import { shallow } from 'enzyme';
import Input from './index';

describe('[Component: Input]', () => {
  let wrapper;
  const props = {
    title: 'Name',
    name: 'name',
    type: 'text',
    required: true,
    validation: 'no-empty',
  };
  beforeEach(() => {
    wrapper = shallow(<Input {...props} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
