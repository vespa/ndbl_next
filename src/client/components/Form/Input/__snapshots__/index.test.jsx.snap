// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Input] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    name="name"
    onChange={[Function]}
    required={true}
    title="Name"
    type="text"
    validation="no-empty"
    value=""
  >
    <span />
  </Input>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <label
        htmlFor="name"
      >
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          Name
        </span>
        <input
          className=" undefined inp inp-name"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          title="Name"
          type="text"
          validation="no-empty"
          value=""
        />
        <span />
      </label>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            style={
              Object {
                "display": "none",
              }
            }
          >
            Name
          </span>,
          <input
            className=" undefined inp inp-name"
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            required={true}
            title="Name"
            type="text"
            validation="no-empty"
            value=""
          />,
          <span />,
        ],
        "htmlFor": "name",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
            "style": Object {
              "display": "none",
            },
          },
          "ref": null,
          "rendered": "Name",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": " undefined inp inp-name",
            "id": "name",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Name",
            "required": true,
            "title": "Name",
            "type": "text",
            "validation": "no-empty",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "label",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <label
          htmlFor="name"
        >
          <span
            style={
              Object {
                "display": "none",
              }
            }
          >
            Name
          </span>
          <input
            className=" undefined inp inp-name"
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            required={true}
            title="Name"
            type="text"
            validation="no-empty"
            value=""
          />
          <span />
        </label>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              style={
                Object {
                  "display": "none",
                }
              }
            >
              Name
            </span>,
            <input
              className=" undefined inp inp-name"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              required={true}
              title="Name"
              type="text"
              validation="no-empty"
              value=""
            />,
            <span />,
          ],
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
              "style": Object {
                "display": "none",
              },
            },
            "ref": null,
            "rendered": "Name",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " undefined inp inp-name",
              "id": "name",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "required": true,
              "title": "Name",
              "type": "text",
              "validation": "no-empty",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "label",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
